src/http-operation.ts

```diff
@@ -0,0 +1,199 @@
+import { INode, INodeExample } from './graph';
+import { ISchema } from './schemas';
+import { IServer } from './servers';
+
+export interface IHttpOperation extends INode {
+  method: string;
+  path: string;
+  responses: IHttpOpResponse[];
+  servers?: IServer[];
+  request?: IHttpOpRequestParams;
+  security?: HttpOpSecurityScheme[];
+  deprecated?: boolean;
+}
+
+// Inspired by: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#parameterObject
+export interface IHttpOpParam {
+  name: string;
+  style?: HttpOpParamStyles;
+  description?: string;
+  required?: boolean;
+  explode?: boolean;
+  deprecated?: boolean;
+
+  // most of the type, params from oas2/oas3 will simply be converted to content with one '*' key
+  content: {
+    [mediaType: string]: IHttpOpContent;
+  };
+}
+
+export const enum HttpOpParamStyles {
+  Simple = 'simple',
+  Matrix = 'matrix',
+  Label = 'label',
+  Form = 'form',
+  SpaceDelimited = 'spaceDelimited',
+  PipeDelimited = 'pipeDelimited',
+  DeepObject = 'deepObject',
+}
+
+export interface IHttpOpPathParam extends IHttpOpParam {
+  // defaults to simple
+  style?: HttpOpParamStyles.Label | HttpOpParamStyles.Matrix | HttpOpParamStyles.Simple;
+}
+
+export interface IHttpOpQueryParam extends IHttpOpParam {
+  // defaults to form
+  style?:
+    | HttpOpParamStyles.Form
+    | HttpOpParamStyles.SpaceDelimited
+    | HttpOpParamStyles.PipeDelimited
+    | HttpOpParamStyles.DeepObject;
+
+  allowEmptyValue?: boolean;
+  allowReserved?: boolean;
+}
+
+export interface IHttpOpHeaderParam extends IHttpOpParam {
+  // defaults to simple
+  style?: HttpOpParamStyles.Simple;
+}
+
+export interface IHttpOpCookieParam extends IHttpOpParam {
+  // defaults to form
+  style?: HttpOpParamStyles.Form;
+}
+
+export interface IHttpOpEncoding {
+  property: string;
+  mediaType?: string;
+  headers?: IHttpOpHeaderParam[];
+  // deafults to form
+  style:
+    | HttpOpParamStyles.Form
+    | HttpOpParamStyles.SpaceDelimited
+    | HttpOpParamStyles.PipeDelimited
+    | HttpOpParamStyles.DeepObject;
+
+  explode?: boolean;
+  allowReserved?: boolean;
+}
+
+export interface IHttpOpContent {
+  mediaType: string;
+  schema?: ISchema;
+  examples?: INodeExample[];
+  encodings?: IHttpOpEncoding[];
+}
+
+// https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#link-object
+export interface IHttpOpLink {
+  operationRef: string;
+  parameters: Array<{
```

Comment:
Not sure whether it is enough to describe link parameters. According to spec (https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#link-object) parameters should have `in` field to indicate where to put them. In out case this could be simply adding `in` field to parameter def, or maybe use IHttpOpParam? The question is whether we can put expression into IHttpOpParam?

Reply:
Yeah good point but I think we can defer this to later. We don't use the link prop for anything right now - when we get to using it we can revisit to make sure structure is correct.